src/modules/Builder/features/ActivityItems/ConditionalPanel/ConditionalPanel.tsx

```diff
@@ -34,9 +34,14 @@ export const ConditionalPanel = ({ condition }: { condition?: ConditionalLogic }
   const { fieldName } = useCurrentActivity();
 
   const items = watch(`${fieldName}.items`);
-  const currentItem = items?.find(
-    (item: ItemFormValues) => getEntityKey(item) === condition?.itemKey,
+  const itemsByEntityKey = items?.reduce(
```

Comment:
could you use `getObjectFromList` from shared/utils

Reply:
done